### PR TITLE
Doc

### PR DESCRIPTION
**Pull request by parcollet**

### CODE REVIEW

File: doc/documentation/manual/triqs/gfs/contents.rst [hunk 1]
```diff
@@ -1,9 +1,11 @@
-**[triqs/gfs]** Green's functions
+Green's functions
 ===================================
```
Comment (Wentzell) [inline doc/documentation/manual/triqs/gfs/contents.rst:2]:
```suggestion
=================
```

File: triqs/gfs/gf/defs.hpp [hunk 2]
```diff
@@ -91,6 +91,24 @@ namespace triqs {
 
     template <typename G> TRIQS_DEPRECATED("use X.target_shape() instead") auto get_target_shape(G const &g) { return g.target_shape(); }
 
+    /**----------------------------------------------------------
+  *  @return  - A new array of zeros of 0 if Gf is matrix/tensor valued.
+  *           - 0 if Gf is scalar valued.
+  *--------------------------------------------------------*/
+
+    template <typename G> auto make_zero_return(G const &g) {
```
Comment (Wentzell) [inline triqs/gfs/gf/defs.hpp:99]: Do we really want a function for this or shouldn't we just write
```c++
triqs::arrays::zeros(g.target_shape())
```
instead?

File: triqs/gfs/gf/gf.hpp [hunk 3]
```diff
@@ -381,17 +359,16 @@ namespace triqs::gfs {
 	*
 	* The assignment resizes the mesh and the data, invalidating all pointers on them.
 	* 
-	* @tparam RHS Type of the right hand side rhs
+	* @tparam RHS  Type of the right hand side rhs. Must model ImmutableGreenFunction concept.
```
Comment (Wentzell) [inline triqs/gfs/gf/gf.hpp]:
```suggestion
	* @tparam RHS  Type of the right hand side rhs. Must model GreenFunction concept.
```

File: triqs/gfs/gf/gf.hpp [hunk 4]
```diff
@@ -381,17 +359,16 @@ namespace triqs::gfs {
...
 	*
 	* 		  RHS can be anything modeling the gf concept TBW
 	* 		  In particular lazy expression with Green functions
 	* @param rhs
 	* @example    triqs/gfs/gf_assign_0.cpp
 	*/
-    template <typename RHS> gf &operator=(RHS &&rhs) {
+    template <typename RHS> gf &operator=(RHS &&rhs) REQUIRES(ImmutableGreenFunction<RHS>::value) {
```
Comment (Wentzell) [inline triqs/gfs/gf/gf.hpp:369]:
```suggestion
    template <typename RHS> gf &operator=(RHS &&rhs) REQUIRES(IsGreenFunction<RHS>::value) {
```

File: triqs/gfs/gf/gf.hpp [hunk 5]
```diff
@@ -340,9 +318,9 @@ namespace triqs::gfs {
...
 
     /** 
-     *  From any object modeling the ImmutableGreenFunction concept.
+     *  From any object modeling the :ref:`concept_GreenFunction`.
      * 
-     *  @tparam G A type modeling ImmutableGreenFunction.
+     *  @tparam G A type modeling :ref:`concept_GreenFunction`.
      *  @param g 
      */
     template <typename G> gf(G const &g) REQUIRES(ImmutableGreenFunction<G>::value) : gf() { *this = g; }
```
Comment (Wentzell) [inline triqs/gfs/gf/gf.hpp:326]:
```suggestion
    template <typename G> gf(G const &g) REQUIRES(IsGreenFunction<G>::value) : gf() { *this = g; }
```

**Review by Wentzell — CHANGES_REQUESTED** [review]

File: triqs/gfs/gf/defs.hpp (hunk 2 above)
Comment (parcollet) [reply triqs/gfs/gf/defs.hpp:99]: It will not work for the scalar case.
But yes, we can simplify the implementation.

Comment (Wentzell) [reply triqs/gfs/gf/defs.hpp:99]: Couldn't we even make that work for the scalar case?

**Comment by Wentzell** [comment]
This was merged into unstable.

**Pull request closed without merging**